src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -916,13 +916,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             userDefinedConversionsMightStillBeApplicable = False
             Dim conv As ConversionKind
 
-            ' Using <symbol>.IsConstant() for field accesses can result in an infinite loop.
+            ' Using source.IsConstant for field accesses can result in an infinite loop.
             ' To detect such a loop pass the already visited constants from the binder.
+            ' The check for source.IsConstant is still necessary because the node might still 
+            ' be considered as a non-constant in some error conditions (a reference before declaration,
+            ' for example).
```

Comment:
This comment now reads as saying that source.IsConstant can cause an infinite loop, but we're going to do it anyway.

I presume checking the `ConstantFieldsInProgress` first allows us to break the cycle. If so, can you please consolidate this comment to explain the best practices around calling `source.IsConstant` in this method?